### ISSUE TITLE
Add all spell events to ClosureEvents

### ISSUE THREAD

**Issue by WareditorGit** [wurstscript/WurstStdlib2#235]
It would be useful to have all those events in ClosureEvents:
EVENT_PLAYER_UNIT_SPELL_CHANNEL
EVENT_PLAYER_UNIT_SPELL_CAST
EVENT_PLAYER_UNIT_SPELL_EFFECT
EVENT_PLAYER_UNIT_SPELL_FINISH
EVENT_PLAYER_UNIT_SPELL_ENDCAST

Issue is that onCast is used for the EVENT_PLAYER_UNIT_SPELL_EFFECT event so we would have to find different names. After a conversation with @Frotty we came up with (in order):
onChannel, onStart, onCast, onFinish and onEnd.
Would be nice to have some feedback/suggestions before doing the PR.

**Comment by Donach** [comment]
@Krypt0n What about pushing BetterCast here?

**Comment by Krypt0n** [comment]
Ok so we have some enhanced casting System here, called "BetterCast" which handles several closure events, while using a preset ability (based on channel) which does mostly nothing but handling cooldown and manacost check.
Current usable events:

```
OnBegin        
OnFail         
OnPreCast      
OnCastStart    
OnCastUpdate   
OnCastTick     
OnCastEnd      
OnEffect       
OnChannelStart 
OnChannelUpdate
OnChannelTick  
OnChannelEnd   
OnEnd          
```

Could do some adaptions and hotdoc, and might prepare it for stdlib, would need some polishing then i guess.

**Comment by Frotty** [comment]
@Krypt0n ❓ 😄

**Comment by Krypt0n** [comment]
Yeah sup, well I was planning to reworked the system, and the thing is scale it a bit higher and that is not dooable by plain wc3 natives. So it will no run with my Custom UI API which is also WIP, and that is way to much for stdlib tbh.

**Comment by Frotty** [comment]
> Yeah sup, well I was planning to reworked the system, and the thing is scale it a bit higher and that is not dooable by plain wc3 natives.

What?

**Comment by Krypt0n** [comment]
As I said. Its not done, and I pause that, cause working now on a Custom UI API, which will be required for the CastingSystem as it does fit all requirements with using, normal wc3 native abilities and triggers.

**Comment by Frotty** [comment]
I don't see how UI is related to adding spell events

**Comment by Donach** [comment]
Everything...

**Comment by Krypt0n** [comment]
Its basically the essence, cause without custom UI you can no clearing trigger spells that nicely and freely without interacting with units orders etc.

**Comment by Frotty** [comment]
> Everything...

No. And then as @Krypt0n already said, this isn't stdlib material.
This ticket is simply about adding spell events,

**Comment by Cokemonkey11** [comment]
@WareditorGit is this still valid?

**Comment by Frotty** [comment]
yes
